client/components/checklist/index.js

```diff
@@ -0,0 +1,72 @@
+/** @format */
+/**
+ * External dependencies
+ */
+import classNames from 'classnames';
+import React, { Children, PureComponent } from 'react';
+import PropTypes from 'prop-types';
+import { times } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import ChecklistHeader from 'components/checklist/header';
+import TaskPlaceholder from 'components/checklist/task-placeholder';
+
+export default class Checklist extends PureComponent {
+	static propTyps = {
+		completedCount: PropTypes.number,
+		inferCompletedCount: PropTypes.bool,
+		isPlaceholder: PropTypes.bool,
+	};
+
+	state = { hideCompleted: false };
+
+	toggleCompleted = () =>
+		this.setState( ( { hideCompleted } ) => ( { hideCompleted: ! hideCompleted } ) );
+
+	renderPlaceholder() {
+		return (
+			<div className={ classNames( 'checklist', 'is-expanded', 'is-placeholder' ) }>
+				<ChecklistHeader total={ 0 } completed={ 0 } />
+				<div className="checklist__tasks">
+					{ times( Children.count( this.props.children ), index => (
+						<TaskPlaceholder key={ index } />
+					) ) }
+				</div>
+			</div>
+		);
+	}
+
+	render() {
+		if ( this.props.isPlaceholder ) {
+			return this.renderPlaceholder();
+		}
+
+		const { children, completedCount, inferCompletedCount } = this.props;
+
+		const count = inferCompletedCount
+			? Children.map( children, child => child.props.completed ).reduce(
+					( acc, completed ) => ( true === completed ? acc + 1 : acc ),
+					0
+			  )
+			: completedCount;
+
+		return (
+			<div
+				className={ classNames( 'checklist', {
+					'is-expanded': ! this.state.hideCompleted,
+					'hide-completed': this.state.hideCompleted,
```

Comment:
Hmm, reading these CSS classes like this, they seem kinda redundant...?

Reply:
I agree, however that's not been introduced here. I took a quick look and cleanup is more involved than I'd like to include in this PR. Let's leave it for now.